go/vt/vtgate/planbuilder/operators/joins.go

```diff
@@ -33,6 +33,10 @@ type JoinOp interface {
 	AddJoinPredicate(ctx *plancontext.PlanningContext, expr sqlparser.Expr)
 }
 
+func IsOuter(outer JoinOp) bool {
+	return !outer.IsInner()
+}
```

Comment:
Would it make sense to define this on the `JoinOp` type? I'm not really sure when we prefer that over a standalone function, so I'm asking mostly out of curiosity. 🙇‍♂️

Reply:
I did consider that, but that forces all implementations to have to have an implementation of this method, and that seems silly.

If golang had some way of defining method implementations on interfaces, I'd use that